plio/signals.py

```diff
@@ -28,3 +28,11 @@ def item_update_cache(sender, instance, **kwargs):
 def question_update_cache(sender, instance, **kwargs):
     # invalidate saved cache for the plio
     invalidate_cache_for_instance(instance.item.plio)
+
+
+@receiver([post_save], sender=Question)
+def question_delete_image_on_deletion(sender, instance, **kwargs):
+    # check if any image is linked to the instance
+    if instance.deleted is not None and instance.image is not None:
+        # delete that image as well
+        instance.image.delete()
```

Comment:
is this soft delete or what? Will the entry get deleted from the DB as well?
If yes, then we should also delete the image on S3 right? Otherwise we'll end up with a lot of images on S3 folder which are not connected to anything. I don't think it's much of a problem but things will be neat and tidy, no loose ends

Reply:
mention soft delete